pennylane_lightning/core/_adjoint_jacobian_base.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Internal methods for adjoint Jacobian differentiation method.
+"""
+
+from os import getenv
+from typing import Any, Callable, List
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, QuantumFunctionError, StatePrep
+from pennylane.measurements import Expectation, MeasurementProcess, State
+from pennylane.operation import Operation
+from pennylane.tape import QuantumTape
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+from pennylane_lightning.core.lightning_base import _chunk_iterable
+
+
+class LightningBaseAdjointJacobian:
+    """Check and execute the adjoint Jacobian differentiation method.
+
+    Args:
+        qubit_state(Lightning[Device]StateVector): State Vector to calculate the adjoint Jacobian with.
+        batch_obs(bool): If serialized tape is to be batched or not.
+    """
+
+    def __init__(self, qubit_state: Any, batch_obs: bool = False) -> None:
+        self._qubit_state = qubit_state
+        self._state = qubit_state.state_vector
+        self._dtype = qubit_state.dtype
+        self._batch_obs = batch_obs
+
+        # Dummy for the C++ bindings
+        self._jacobian_lightning: Callable = None
+        self._create_ops_list_lightning: Callable = None
+
+    @property
+    def qubit_state(self):
+        """Returns a handle to the Lightning[Device]StateVector object."""
+        return self._qubit_state
+
+    @property
+    def state(self):
+        """Returns a handle to the Lightning internal data object."""
+        return self._state
+
+    @property
+    def dtype(self):
+        """Returns the simulation data type."""
+        return self._dtype
+
+    def _adjoint_jacobian_dtype(self):
```

Comment:
https://docs.python.org/3/library/abc.html#abc.abstractmethod

```suggestion
    @abstractmethod
    def _adjoint_jacobian_dtype(self):
```